diff -ruN ../src/.gitignore .gitignore
--- ../src/.gitignore	1970-01-01 02:00:00.000000000 +0200
+++ .gitignore	2023-02-26 11:58:05.000000000 +0200
@@ -0,0 +1,2 @@
+*
+!.gitignore
diff -ruN ../src/contracts/gho/GhoToken.sol contracts/gho/GhoToken.sol
--- ../src/contracts/gho/GhoToken.sol	2023-02-26 10:23:14.000000000 +0200
+++ contracts/gho/GhoToken.sol	2023-02-26 13:26:13.000000000 +0200
@@ -75,11 +75,16 @@
     uint128 bucketCapacity
   ) external onlyRole(FACILITATOR_MANAGER) {
     Facilitator storage facilitator = _facilitators[facilitatorAddress];
+    require(
+      !facilitator.isLabelNonempty, //TODO: remove workaroun when CERT-977 is resolved
+      'FACILITATOR_ALREADY_EXISTS'
+    );
     require(bytes(facilitator.label).length == 0, 'FACILITATOR_ALREADY_EXISTS');
     require(bytes(facilitatorLabel).length > 0, 'INVALID_LABEL');
 
     facilitator.label = facilitatorLabel;
     facilitator.bucketCapacity = bucketCapacity;
+    facilitator.isLabelNonempty = true;
 
     _facilitatorsList.add(facilitatorAddress);
 
@@ -93,6 +98,10 @@
   /// @inheritdoc IGhoToken
   function removeFacilitator(address facilitatorAddress) external onlyRole(FACILITATOR_MANAGER) {
     require(
+      _facilitators[facilitatorAddress].isLabelNonempty, //TODO: remove workaroun when CERT-977 is resolved
+      'FACILITATOR_DOES_NOT_EXIST'
+    );
+    require(
       bytes(_facilitators[facilitatorAddress].label).length > 0,
       'FACILITATOR_DOES_NOT_EXIST'
     );
@@ -112,6 +121,10 @@
     address facilitator,
     uint128 newCapacity
   ) external onlyRole(BUCKET_MANAGER) {
+    require(
+      _facilitators[facilitator].isLabelNonempty, //TODO: remove workaroun when CERT-977 is resolved
+      'FACILITATOR_DOES_NOT_EXIST'
+    );
     require(bytes(_facilitators[facilitator].label).length > 0, 'FACILITATOR_DOES_NOT_EXIST');
 
     uint256 oldCapacity = _facilitators[facilitator].bucketCapacity;
@@ -126,12 +139,12 @@
   }
 
   /// @inheritdoc IGhoToken
-  function getFacilitatorBucket(address facilitator) external view returns (uint256, uint256) {
+  function getFacilitatorBucket(address facilitator) public view returns (uint256, uint256) {
     return (_facilitators[facilitator].bucketCapacity, _facilitators[facilitator].bucketLevel);
   }
 
   /// @inheritdoc IGhoToken
-  function getFacilitatorsList() external view returns (address[] memory) {
+  function getFacilitatorsList() public view returns (address[] memory) {
     return _facilitatorsList.values();
   }
 }
diff -ruN ../src/contracts/gho/interfaces/IGhoToken.sol contracts/gho/interfaces/IGhoToken.sol
--- ../src/contracts/gho/interfaces/IGhoToken.sol	2023-02-26 10:23:14.000000000 +0200
+++ contracts/gho/interfaces/IGhoToken.sol	2023-02-26 11:58:05.000000000 +0200
@@ -14,6 +14,7 @@
     uint128 bucketCapacity;
     uint128 bucketLevel;
     string label;
+    bool isLabelNonempty;   //TODO: remove workaroun when https://certora.atlassian.net/browse/CERT-977 is resolved
   }
 
   /**
